src/ColumnDisplayFormatDialog.cpp

```diff
@@ -10,29 +10,59 @@ ColumnDisplayFormatDialog::ColumnDisplayFormatDialog(const QString& colname, QSt
-    ui->comboDisplayFormat->addItem(tr("Julian day to date"), "julian");
-    ui->comboDisplayFormat->addItem(tr("Lower case"), "lower");
     ui->comboDisplayFormat->addItem(tr("Octal number"), "octal");
     ui->comboDisplayFormat->addItem(tr("Round number"), "round");
+    ui->comboDisplayFormat->insertSeparator(ui->comboDisplayFormat->count());
+    ui->comboDisplayFormat->addItem(tr("Apple NSDate to date"), "appleDate");
+    ui->comboDisplayFormat->addItem(tr("Java epoch (milliseconds) to date"), "javaEpoch");
+    ui->comboDisplayFormat->addItem(tr("Julian day to date"), "julian");
     ui->comboDisplayFormat->addItem(tr("Unix epoch to date"), "epoch");
-    ui->comboDisplayFormat->addItem(tr("Upper case"), "upper");
     ui->comboDisplayFormat->addItem(tr("Windows DATE to date"), "winDate");
+    ui->comboDisplayFormat->insertSeparator(ui->comboDisplayFormat->count());
+    ui->comboDisplayFormat->addItem(tr("Lower case"), "lower");
+    ui->comboDisplayFormat->addItem(tr("Upper case"), "upper");
     ui->labelDisplayFormat->setText(ui->labelDisplayFormat->text().arg(column_name));
 
+    formatFunctions["lower"] = "lower(" + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["upper"] = "upper(" + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["epoch"] = "datetime(" + sqlb::escapeIdentifier(column_name) + ", 'unixepoch')";
+    formatFunctions["javaEpoch"] = "strftime('%Y-%m-%d %H:%M:%S.', " + sqlb::escapeIdentifier(column_name) +
+      "/1000, 'unixepoch') || (" + sqlb::escapeIdentifier(column_name) + "%1000)";
+    formatFunctions["winDate"] = "datetime('1899-12-30', " + sqlb::escapeIdentifier(column_name) + " || \" days\")";
+    formatFunctions["appleDate"] = "datetime('2001-01-01', " + sqlb::escapeIdentifier(column_name) + " || \" seconds\")";
+    formatFunctions["julian"] = "datetime(" + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["round"] = "round(" + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["hex"] = "printf('0x%x', " + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["octal"] = "printf('%o', " + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["exponent"] = "printf('%e', " + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["hexblob"] = "hex(" + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["decimal"] = "printf('%d', " + sqlb::escapeIdentifier(column_name) + ")";
+
     // Set the current format, if it's empty set the default format
     if(current_format.isEmpty())
     {
         ui->comboDisplayFormat->setCurrentIndex(0);
         updateSqlCode();
     } else {
-        ui->comboDisplayFormat->addItem(tr("Custom"), "custom");
-        ui->comboDisplayFormat->setCurrentIndex(ui->comboDisplayFormat->findData("custom"));
+        QString formatName;
+	foreach (const QString &formatKey, formatFunctions.keys()) {
```

Comment:
You can also use a [range-based for loop](https://en.wikipedia.org/wiki/C%2B%2B11#Range-based_for_loop) from C++11 here since we don't have to support older compilers anymore. The ```foreach``` provided by Qt is slower than the C++11 one because it creates a copy of the entire map before iterating over the copy. It's not a big deal though in this case.

Reply:
Perfect. I've committed the improvement.